Fix GTK for Windows links in README

rnestler opened the pull request:
Fixes issue #216

GuillaumeGomez commented on README.md:
```diff
@@ -32,7 +32,7 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
 
 Install [mingw-w64](http://mingw-w64.yaxm.org/) (select the win32 threading model) and download a __GTK+__ SDK:
- * The GNOME project has an official distribution of GTK+ 3.6: [x86](http://www.gtk.org/download/win32.php), [x64](http://www.gtk.org/download/win64.php).
+ * The GNOME project has official distributions of GTK+ 3.4 upto 3.10: http://win32builder.gnome.org/
```
I'd prefer if the link was put "inside" "3.4 upto 3.10". Example:

The GNOME project has official distributions of GTK+ [3.4 upto 3.10](http://win32builder.gnome.org/).

rnestler replied:
What about "The GNOME project has [official distributions](http://win32builder.gnome.org/) of GTK+ 3.4 upto 3.10" ?

GuillaumeGomez replied:
Oh perfect!

GuillaumeGomez commented:
Thanks! Once we confirmed my note, I'll merge.

rnestler commented:
OK I updated the PR like we discussed, should be ready to merge.

GuillaumeGomez commented:
Didn't you want to add a howto as well ?

rnestler commented:
Ah sorry, didn't see your comment on #218 until now. Yeah I con do it in this PR as well.

GuillaumeGomez commented:
Then ping me when it's done please, github doesn't notify when a PR has been updated.

rnestler commented:
OK I added a first version of the HowTo. A few points to note:
 - Should we keep the old instructions or should I just remove them?
 - I only tested in a 32Bit Windows VM but wrote the HowTo for 64Bit since I guessed it's more common.
 - I also added a note about installing rust for the GNU ABI and not installing "Linker and plattform libraries".

gkoz commented on README.md:
```diff
@@ -31,17 +31,47 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+![Screenshot](rust_setup.png)
+
+If you're not sure, check if your Rust installation has `gcc.exe` and `ld.exe`
+in its `bin` directory, you may get a linking error `ld: cannot find -limm32`.
```
This sentence seems slightly broken now. And I think it's now safe to say that the presence of those files will definitely cause the error.

rnestler replied:
Yes you're right. Will fix that.

gkoz commented on README.md:
```diff
@@ -31,17 +31,47 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+
 Install [mingw-w64](http://mingw-w64.yaxm.org/) (select the win32 threading model) and download a __GTK+__ SDK:
- * The GNOME project has an official distribution of GTK+ 3.6: [x86](http://www.gtk.org/download/win32.php), [x64](http://www.gtk.org/download/win64.php).
+ * The GNOME project has [official distributions](http://win32builder.gnome.org/) of GTK+ 3.4 upto 3.10
```
They're not official any more, let's just say they still host some binaries but recommend msys2.

gkoz commented:
> Should we keep the old instructions or should I just remove them?

It makes sense to remove them but maybe later after we have some feedback about the msys2 way.

> I only tested in a 32Bit Windows VM but wrote the HowTo for 64Bit since I guessed it's more common.

I can test this on 64-bit later. BTW you don't need a local VM, appveyor [allows rdp access](https://www.appveyor.com/docs/how-to/rdp-to-build-worker).

gkoz commented on README.md:
```diff
@@ -31,17 +31,47 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+
+#### 1. msys2 toolchain
+
+This method is recommended according to the [gtk project]
```
`gtk` should probably be spelled `GTK+` here.

rnestler commented:
> I can test this on 64-bit later. BTW you don't need a local VM, appveyor allows rdp access.

Ah that's awesome I didn't now that! :smile: This can safe me a lot of time while trying to get [this](https://github.com/rnestler/rust-torrent-search/pull/10) to build with AppVeyor!

rnestler commented:
@gkoz I tried to address all your points. Tell me if I should squash the commits or if there are any other issues.

GuillaumeGomez commented:
Please keep the first commit and squash other in the second one please.

rnestler commented:
@GuillaumeGomez Squashed it. Is it OK like that?

GuillaumeGomez commented:
@rnestler: Perfect! But it's @gkoz who will merge when he thinks everything's ok.

gkoz commented:
Some grammar nits...

gkoz commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+"Linker and platform libraries" in the Rust setup.
+![Screenshot](rust_setup.png)
+
+If you already installed Rust, check if your Rust installation has `gcc.exe`
```
"If you already have installed"

gkoz commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+This method is recommended according to the [GTK+ project]
+(http://www.gtk.org/download/windows.php).  You can follow [this guide]
+(https://blogs.gnome.org/nacho/2014/08/01/how-to-build-your-gtk-application-on-windows/)
+or follow this steps:
```
"these steps"

rnestler commented:
> Some grammar nits...

Yeah sorry about that. I'm not a native english speaker and my grammar is poor :cry:

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+![Screenshot](rust_setup.png)
+
+If you already installed Rust, check if your Rust installation has `gcc.exe`
+and `ld.exe` in its `bin` directory.  In that case remove those executables,
```
"  In" -> " In"

rnestler replied:
I assume you mean I should remove the double space? That's actually my vim setup auto formatting the sentences with two spaces after the period.

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+
+If you already installed Rust, check if your Rust installation has `gcc.exe`
+and `ld.exe` in its `bin` directory.  In that case remove those executables,
+they will be provided by mingw instead.  If you won't you'll probably get a
```
"  If" -> " If"

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
 
+Make sure you have the [GNU
+ABI](https://www.rust-lang.org/downloads.html#win-foot) version of the rust
+compiler installed.  Since we need our own toolchain make sure to uncheck
```
"  Since" -> " Since"

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+#### 1. msys2 toolchain
+
+This method is recommended according to the [GTK+ project]
+(http://www.gtk.org/download/windows.php).  You can follow [this guide]
```
"  You" -> " You"

You have an issue with extra whitespace.

gkoz replied:
Well, it's not like the extra spaces are going to be rendered ;)

GuillaumeGomez replied:
It hurts my eyes. :'(

rnestler replied:
No problem I can fix it :wink:

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+
+ - Install and update [msys2](https://msys2.github.io/)
+ - Install the mingw toolchain
+ ```
````
Add an empty line before "```".

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+ ```
+ $ pacman -S mingw-w64-x86_64-toolchain
+ ```
+ - Install the gtk3 package
````
Add an empty line before "- Install the..."

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+ $ pacman -S mingw-w64-x86_64-toolchain
+ ```
+ - Install the gtk3 package
+ ```
````
Add an empty line before "```".

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
 Install [mingw-w64](http://mingw-w64.yaxm.org/) (select the win32 threading model) and download a __GTK+__ SDK:
- * The GNOME project has an official distribution of GTK+ 3.6: [x86](http://www.gtk.org/download/win32.php), [x64](http://www.gtk.org/download/win64.php).
+ * The GNOME project hosts [distributions](http://win32builder.gnome.org/) of GTK+ 3.4 upto 3.10
  * [GTK+ for Windows Runtime Environment Installer: 64-bit](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer) supports GTK+ 3.14, its SDK downloads can currently be found [here](http://lvserver.ugent.be/gtk-win64/sdk/).
```
"its SDK downloads" -> "its SDK download links"

rnestler replied:
Actually this is not from me but I can change it anyway :wink:

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,48 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
  * [GTK+ for Windows Runtime Environment Installer: 64-bit](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer) supports GTK+ 3.14, its SDK downloads can currently be found [here](http://lvserver.ugent.be/gtk-win64/sdk/).
 
 Make sure both mingw's and the sdk's `bin` directories are in your `PATH` e.g. (assuming mingw is installed in `C:\mingw-w64` and the SDK unpacked into `C:\gtk`)
 ```
````
Add an empty line before "```".

gkoz commented:
Looks good :+1:

GuillaumeGomez commented on README.md:
```diff
@@ -31,17 +31,52 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+or follow these steps:
+
+ - Install and update [msys2](https://msys2.github.io/)
+ - Install the mingw toolchain
```
Please add ":" at the end of this sentence.

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,52 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+ - Install and update [msys2](https://msys2.github.io/)
+ - Install the mingw toolchain
+
+ ```
````
"```Shell" would be preferable to avoid potential github automatic highlighting error.

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,52 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+
+ - Install the gtk3 package
+
+ ```
````
"```Shell" again.

GuillaumeGomez commented on README.md:
````diff
@@ -31,17 +31,52 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+ $ pacman -S mingw-w64-x86_64-toolchain
+ ```
+
+ - Install the gtk3 package
````
Add ":" at the end of the sentence.

GuillaumeGomez commented:
Seems good to me too. Once you have fixed my last notes, we merge.

rnestler commented:
@GuillaumeGomez Done :)

GuillaumeGomez commented:
Thanks a lot for your contribution @rnestler! I merge.

rnestler commented:
Happy to help! Thanks for your work on gtk-rs!

gkoz commented on README.md:
````diff
@@ -31,17 +31,52 @@ __gtk__ expects __GTK+__, __GLib__ and __Cairo__ development files to be install
+ - Install the mingw toolchain:
+
+ ```Shell
+ $ pacman -S mingw-w64-x86_64-toolchain
````
Does this actually work? I've tried this in appveyor and it didn't because the msys2 mingw is incorrect flavor, it's not win32+dwarf/seh. I ended up [using mingw-builds](https://github.com/gkoz/gtk-rs-gtk/commit/faab1070353fbb01f855eecc833bb8592957672e), the installer linked at http://mingw-w64.org/doku.php/download/mingw-builds

gkoz replied:
I wasn't using the msys2-provided gtk though but the same distributions as before.
